client/app/bundles/HelloWorld/components/NewResturantForm.jsx

```diff
@@ -0,0 +1,181 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+export default class NewResturantForm extends React.Component {
+
+    static propTypes = {
+        cuisines: PropTypes.array.isRequired
+    };
+
+    /*componentDidMount() {
+        console.log('mounted');
+        navigator.geolocation.getCurrentPosition((position) => {
+            console.log(position)
+        })
+    }*/
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            name: '',
+            cuisine: '',
+            tenbis: false,
+            address: '',
+            maxDeliveryTime: ''
+        }
+    };
+
+    newResturantSubmit = (e) => {
+        e.preventDefault();
+        $.ajax({
+            url: '/resturants',
+            dataType: 'json',
+            type: 'POST',
+            data: {resturant: {name: this.state.name,
+                tenbis: this.state.tenbis,
+                address: this.state.address,
+                max_delivery_time: this.state.maxDeliveryTime,
+                cuisine_id: this.state.cuisine},
+            },
+            success: function() {
+                window.location.href = "/";
```

Comment:
Well, typically in a single page application (SPA) you'd want to have dynamic pages (see React Router) where the application is packed in a single bundle and it's decoupled from the Rails Application what-so-ever. Then, after a successful XHR request you'd append the new data to your data-store and hence achieve an update-to-date state - all without reloading the application. In this scenario, we'd usually use a datastore such as Redux which propagates throughout the application once updated (reduced).

See https://scotch.io/tutorials/routing-react-apps-the-complete-guide